public/app/features/dashboard/api/dashboard_api.ts

```diff
@@ -2,45 +2,43 @@ import { DashboardV2Spec } from '@grafana/schema/dist/esm/schema/dashboard/v2alp
 import { DashboardDTO } from 'app/types';
 
 import { LegacyDashboardAPI } from './legacy';
+import { TransitionalDashboardAPI } from './transitional_dashboard_api';
 import { DashboardAPI, DashboardWithAccessInfo } from './types';
 import { getDashboardsApiVersion } from './utils';
 import { K8sDashboardAPI } from './v0';
 import { K8sDashboardV2APIStub } from './v2';
 
-// Describes the dashboard DTO types per API version
-export interface ApiVersionDTO {
-  legacy: DashboardDTO;
-  v0: DashboardDTO;
-  // v1: DashboardDTO; TODO[schema]: enable v1 when available
-  v2: DashboardWithAccessInfo<DashboardV2Spec>;
-}
+type DashboardAPIClients = {
+  legacy: DashboardAPI<DashboardDTO>;
+  v0: DashboardAPI<DashboardDTO>;
+  v2: DashboardAPI<DashboardDTO>;
+};
 
-type DashboardAPIClients = Record<
-  keyof ApiVersionDTO,
-  DashboardAPI<DashboardDTO | DashboardWithAccessInfo<DashboardV2Spec>>
->;
+type V2ModeOptions = {
+  useV2Mode: true;
+};
 
 let clients: Partial<DashboardAPIClients> | undefined;
 
-export function setDashboardAPI(override: Partial<DashboardAPIClients> | undefined) {
-  if (process.env.NODE_ENV !== 'test') {
-    throw new Error('dashboardAPI can be only overridden in test environment');
-  }
-  clients = override;
-}
+// Overloads
+export function getDashboardAPI(): DashboardAPI<DashboardDTO>;
+export function getDashboardAPI(opts: V2ModeOptions): DashboardAPI<DashboardWithAccessInfo<DashboardV2Spec>>;
+
+export function getDashboardAPI(opts?: V2ModeOptions): DashboardAPI<any> {
```

Comment:
Can this be simply a boolean? or a string `v2`? I think it's more descriptive and simpler to use than an object.